source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -31,8 +60,37 @@ Api::IoCallBoolResult FileImplWin32::open(FlagSet in) {
     return resultSuccess(true);
   }
 
-  auto flags = translateFlag(in);
-  fd_ = CreateFileA(path_.c_str(), flags.access_, FILE_SHARE_READ | FILE_SHARE_WRITE, 0,
+  auto std_handle = toStandardHandle(path_);
```

Comment:
nit: maybe just add some comments in this section to ensure its easy to understand with a quick glance, should already be pretty self explanatory but for those that are not familiar

Reply:
please use an explicit type here, also per style guide.